mindsdb/integrations/handlers/lancedb_handler/lancedb_handler.py

```diff
@@ -0,0 +1,372 @@
+import json
+
+from mindsdb.integrations.libs.const import HANDLER_CONNECTION_ARG_TYPE as ARG_TYPE
+from mindsdb.integrations.libs.response import RESPONSE_TYPE
+from mindsdb.integrations.libs.response import HandlerResponse
+from mindsdb.integrations.libs.response import HandlerResponse as Response
+from mindsdb.integrations.libs.response import HandlerStatusResponse as StatusResponse
+from mindsdb.integrations.libs.vectordatabase_handler import (
+    FilterCondition,
+    FilterOperator,
+    TableField,
+    VectorStoreHandler,
+)
+from mindsdb.utilities import log
+
+
+class LanceDBHandler(VectorStoreHandler):
+    """This handler handles connection and execution of the LanceDB statements."""
+
+    name = "lancedb"
+
+    def __init__(self, name: str, **kwargs):
+        super().__init__(name, **kwargs)
+        self._connection_data = kwargs.get("connection_data")
+
+        self._client_config = {
+            "uri": self._connection_data.get("persist_directory"),
+            "api_key": self._connection_data.get("api_key", None),
+            "region": self._connection_data.get("region"),
+            "host_override": self._connection_data.get("host_override"),
+        }
+
+        # uri is required either for LanceDB Cloud or local
+        if not self._client_config["uri"]:
+            raise Exception(
+                "persist_directory is required for LanceDB connection!"
+            )
+        # uri, api_key and region is required either for LanceDB Cloud
+        elif self._client_config["uri"] and self._client_config["api_key"] and not self._client_config["region"]:
+            raise Exception(
+                "region is required for LanceDB Cloud connection!"
+            )
+
+        self._client = None
+        self.is_connected = False
+        self.connect()
+
+    def _get_client(self):
+        client_config = self._client_config
+        if client_config is None:
+            raise Exception("Client config is not set!")
+        return lancedb.connect(**client_config)
+
+    def __del__(self):
+        if self.is_connected is True:
+            self.disconnect()
+
+    def connect(self):
+        """Connect to a LanceDB database."""
+        if self.is_connected is True:
+            return
+        try:
+            self._client = self._get_client()
+            self.is_connected = True
+        except Exception as e:
+            log.logger.error(f"Error connecting to LanceDB client, {e}!")
+            self.is_connected = False
+
+    def disconnect(self):
+        """Close the database connection."""
+        if self.is_connected is False:
+            return
+        self._client = None
+        self.is_connected = False
+
+    def check_connection(self):
+        """Check the connection to the LanceDB database."""
+        response_code = StatusResponse(False)
+        need_to_close = self.is_connected is False
+
+        try:
+            # self._client.heartbeat()
+            response_code.success = True
```

Comment:
This doesn't check the connection? Do we need the heartbeat uncommented?

Reply:
Hi @ZoranPandovski there is no self._client.heartbeat() in lancedb. I fixed that piece and pushed.